src/parsers/index.js

```diff
@@ -0,0 +1,147 @@
+            transactionsTable.field.newNextForgingKeyIndex,
+            transactionsTable.field.newNextMultisigKeyIndex,
+            transactionsTable.field.newNextSigKeyIndex,
+        ];
+
+        const base64Fields = [
+            transactionsTable.field.signatures,
+        ];
+        const textArrayFields = [
+            transactionsTable.field.memberAddresses,
+        ];
+
+        let parsers = [
+            sanitizeTransaction,
+            (txn) => numberParser(txn, bigIntegerFields),
+            (txn) => base64ObjParser(txn, base64Fields),
+            (txn) => textToArray(txn, textArrayFields)
+        ]
+
+        if (this.knexClient.isSqliteClient()) {
+            const integerFields = [
+                transactionsTable.field.indexInBlock,
+                transactionsTable.field.requiredSignatureCount
+            ]
+            parsers.push((txn) => numberParser(txn, integerFields))
+        }
+        return parsers
+    };
+
+    generateBlocksTableParser = () => {
+        const bigIntegerFields = [
+            blocksTable.field.height,
+            blocksTable.field.timestamp,
+            blocksTable.field.nextForgingKeyIndex,
+        ];
+
+        const base64Fields = [
+            blocksTable.field.signatures,
+        ];
+
+        let parsers = [
+            (block) => numberParser(block, bigIntegerFields),
+            (block) => base64ObjParser(block, base64Fields),
+            removePrivateBlockField
+        ]
+
+        if (this.knexClient.isSqliteClient()) {
+            const booleanFields = [
+                blocksTable.field.active
+            ]
+            const integerFields = [
+                blocksTable.field.numberOfTransactions
+            ]
+            parsers.push(
+                (block) => numberParser(block, integerFields),
+                (block) => booleanParser(block, booleanFields)
+            )
+        }
+        return parsers
+    };
+
+    generateDelegatesTableParser = () => {
+        const bigIntegerFields = [delegatesTable.field.updateHeight];
+        return [
+            (delegate) => numberParser(delegate, bigIntegerFields)
+        ]
+    };
+
+    generateBallotsTableParser = () => {
+        let parsers = []
+        if (this.knexClient.isSqliteClient()) {
+            const booleanFields = [
+                ballotsTable.field.active
+            ]
+            parsers.push(
+                (ballot) => booleanParser(ballot, booleanFields),
+            )
+        }
+        return parsers
+    };
+
+    initializeParsers = () => {
+        this.accountTableParsers = this.generateAccountsTableParsers()
+        this.transactionsTableParsers = this.generateTransactionTableParsers()
+        this.blocksTableParsers = this.generateBlocksTableParser()
+        this.delegatesTableParsers = this.generateDelegatesTableParser()
+        this.ballotsTableParsers = this.generateBallotsTableParser()
+    }
+
+    getAppliedParsers = () => {
```

Comment:
The name `getAppliedParsers` sounds technical (sounds like a private method). How about something more meaningful like `getRecordParsers`? Or `getResultParsers`?

Reply:
Updated to recorded parsers